### PR TITLE
TDL 16573 sync teams at organization level

### PR DESCRIPTION
**Pull request by prijendev**
# Description of change
- Sync `teams`, `team_members`, and `team_memberships` single time only for each organization.
- Sync all other streams for every repo.
# Manual QA steps
 - Provide  multiple repos with common organization and verify that the above 3 streams sync only 1 time.
 - Provide  multiple repos with common organization and verify that all streams except the above 3 streams sync for each repo.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_github/sync.py [hunk 1]
```diff
@@ -106,6 +106,15 @@ def write_schemas(stream_id, catalog, selected_streams):
     for child in stream_obj.children:
         write_schemas(child, catalog, selected_streams)
 
+def get_stream_to_sync_for_orgs(streams_to_sync):
+    """
+    Return set of streams to sync among the `teams`, 'team_members' and 'team_memberships'.
+    """
+    streams_to_sync_for_orgs = ['teams', 'team_members', 'team_memberships']
+    streams_to_sync_for_orgs = [stream for stream in streams_to_sync if stream in streams_to_sync_for_orgs]
```
[the remark this answers is not in the document]
Comment (prijendev) [reply tap_github/sync.py]: Updated.

File: tap_github/client.py [hunk 2]
```diff
@@ -224,6 +224,15 @@ def verify_access_for_repo(self):
             # Verifying for Repo access
             self.verify_repo_access(url_for_repo, repo)
 
+    def extract_orgs_from_config(self):
```
Comment (dbshah1212) [inline tap_github/client.py:227]: I guess you should do code on https://github.com/singer-io/tap-github/pull/166 because what if provided org is invalid?

Comment (prijendev) [reply tap_github/client.py:240]: Updated.

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

**Comment by sgandhi1311** [comment]
Changes looks good to me. Can you please resolve the merge conflicts on files - `test_extract_repos_from_config.py` and `test_stream.py` ?

**Comment by prijendev** [comment]
> Changes looks good to me. Can you please resolve the merge conflicts on files - `test_extract_repos_from_config.py` and `test_stream.py` ?

Resolved merge conflict.

**Pull request merged**